network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt

```diff
@@ -35,7 +38,15 @@ interface WgServerApi {
         val gateway: String,
     )
 
-    suspend fun registerPublicKey(publicKey: String): WgServerData?
+    suspend fun registerPublicKey(
+        publicKey: String,
+        mode: Mode = Default,
+    ): WgServerData?
+
+    sealed class Mode {
+        data object Default : Mode()
+        data class FailureRecovery(val currentServer: String) : Mode()
```

Comment:
Why not overload the `toString` method (like below) so that we don't need to do the mapping between the model and the `MODE_FAILURE` constant and we just do `toString()` on it?

```suggestion
        data class FailureRecovery(val currentServer: String) : Mode() {
            override fun toString(): String {
                return "failureRecovery"
            }
        }
```